src/Denormalizer/AssociationTypeCreatedDenormalizer.php

```diff
@@ -13,26 +13,18 @@
      */
     public function supports(AMQPMessage $message)
     {
-        $decodedMessage = json_decode($message->getBody(), true);
+        $body = json_decode($message->getBody(), true);
 
-        if (
-            isset($decodedMessage['type']) &&
-            MessageType::ASSOCIATION_TYPE_CREATED_MESSAGE_TYPE === $decodedMessage['type'] &&
-            $message->get('content_type') === 'json'
-        ) {
-            return true;
-        }
-
-        return false;
+        return isset($body['type'], $body['payload']) && MessageType::ASSOCIATION_TYPE_CREATED_MESSAGE_TYPE === $body['type'];
     }
 
     /**
      * {@inheritdoc}
      */
     public function denormalize(AMQPMessage $message)
     {
-        $decodedMessage = json_decode($message->getBody(), true);
+        $payload = json_decode($message->getBody(), true)['payload'];
 
-        return new AssociationTypeCreated($decodedMessage['code'], $decodedMessage['labels']);
+        return new AssociationTypeCreated($payload['code'], $payload['labels']);
```

Comment:
Tests incoming, thinking of introducing OptionsResolver here in the next iteration.